### ISSUE TITLE
missing `${BASEHREF}` prefix on several links in the UI

### ISSUE THREAD

**Issue by Guillermogsjc** [argoproj/argo-workflows#9764]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

Hi, very small bug:

UI **OPEN WORKFLOW TEAMPLATE** link is missing `--basehref` settled prefix.  It it will send you to

`workflow-templates/argo/${TEMPLATE_NAME}`

instead of 

`${BASEHREF}/workflow-templates/argo/${TEMPLATE_NAME}`

### Version

v3.4.1

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
no wf
```

### Logs from the workflow controller

no log

### Logs from in your workflow's wait container

no log

**Comment by Guillermogsjc** [comment]
in fact this bug is in the redirecting URL from `${BASEHREF}/argo-workflows/workflows/argo/${WORKFLOW_ID}`, 

that goes to `archived-workflows/argo/${ARCHIVED_WORKFLOW_ID}` instead to 

`${BASEHREF}/archived-workflows/argo/${ARCHIVED_WORKFLOW_ID}`

looks like there is needed a walk over there to check all ${BASEHREF} prefixes on generated urls over the UI

**Comment by Guillermogsjc** [comment]
sorry pulled repo and searched everywhere from `ArgoServerOpts` but did not manage to understand how it is added this basehref to base url before paths

**Comment by hustclf** [comment]
> in fact this bug is in the redirecting URL from `${BASEHREF}/argo-workflows/workflows/argo/${WORKFLOW_ID}`,
> 
> that goes to `archived-workflows/argo/${ARCHIVED_WORKFLOW_ID}` instead to
> 
> `${BASEHREF}/archived-workflows/argo/${ARCHIVED_WORKFLOW_ID}`
> 
> looks like there is needed a walk over there to check all ${BASEHREF} prefixes on generated urls on the UI

hi guys, the issue still exists in v3.4.11. and  I tried to fix it with code below, and it works. cc @Guillermogsjc 
```
--- a/ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx
+++ b/ui/src/app/archived-workflows/components/archived-workflow-list/archived-workflow-list.tsx
@@ -80,7 +80,7 @@ export class ArchivedWorkflowList extends BasePage<RouteComponentProps<any>, Sta
     public componentDidUpdate(): void {
         if (this.state.deep === true && this.state.workflows && this.state.workflows.length === 1) {
             const workflow = this.state.workflows[0];
-            const url = uiUrl('archived-workflows/' + workflow.metadata.namespace + '/' + (workflow.metadata.uid || ''));
+            const url = '/archived-workflows/' + workflow.metadata.namespace + '/' + (workflow.metadata.uid || '');
             this.props.history.push(url);
         }
     }
```

But when I try to open a pr for it,  It seems that `archived-workflow-list/archived-workflow-list.tsx` has beed deleted in latest master for another big feature, refer to : https://github.com/argoproj/argo-workflows/pull/11121

does we still need a hotfix for the latest release (v3.4.11) ? @tczhao

**Comment by Guillermogsjc** [comment]
Hi, I did not hit this anymore since closed and solved, and now have everything into v3.5 so can not verify, but do not remember to have seen it anymore since closed, including V3.4.11

**Comment by agilgur5** [comment]
@hustclf you were correct that this was missing from the Archived Workflows components that were removed for 3.5.0 in #11121, nice catch!

That seems to have been the only place missing when I did a search. I raised a PR to fix it in the 3.4.x patch series in #12172
